# Ceneo Analyzer

## Biblioteki

In [31]:
import os
import pandas as pd

## Lista kodów wszystkich produktów, o których zostały pobrane opinie

In [38]:
print(*[filename.split(".")[0] for filename in os.listdir("opinions")], sep="\n")

156802519
94103927


## Kod produktu, dla którego ma zostać wykonana analiza opinii 94103927

In [45]:
product_id = input("Podaj kod produktu w serwisie Ceneo.pl: ")

## Wczytanie do obiektu pandas.DataFrame opinii o wskazanym produkcie

In [46]:
opinions = pd.read_json(f"opinions/{product_id}.json")

In [47]:
opinions

opinion_id            author recommendation rating  \
0      18387464            MIODAR        Polecam    5/5   
1      18537596          kasiulek        Polecam    5/5   
2      18359374           Natalia        Polecam    5/5   
3      18394372   angelikasitek26        Polecam    5/5   
4      18590279             m...0        Polecam    5/5   
..          ...               ...            ...    ...   
128    18002927  Użytkownik Ceneo        Polecam    5/5   
129    18057101  Użytkownik Ceneo        Polecam    5/5   
130    18022873           Dmitrij        Polecam    5/5   
131    17903116  Użytkownik Ceneo        Polecam    5/5   
132    17947654  Użytkownik Ceneo        Polecam    5/5   

                                               content pros cons  useful  \
0    Jest to mój pierwszy tego typu blender. Postan...   []   []       0   
1    Rewelacyjny, piękny dizajn, posiadam kolor mie...   []   []       0   
2    Dopiero zaczynam przygodę z Blendy, ale:\nJest...   []   []       0   
3    Produkt bardzo solidnie wykonany. Piękny żywy ...   []   []       0   
4    Bardzo fajnie urządzenie, w podróży nie trzeba...   []   []       0   
..                                                 ...  ...  ...     ...   
128  Czekałem na premierę wersji 3 i się nie zawiod...   []   []       0   
129  To mój pierwszy blender od Blendy go i jest to...   []   []       0   
130  blender jest świetny. Szukałem przenośnego. Do...   []   []       0   
131                                        Jest super😀   []   []       0   
132                                            Polecam   []   []       0   

     useless         publish_date        purchase_date  
0          0  2024-02-11 04:05:41  2024-02-03 10:38:15  
1          0  2024-03-30 08:31:50  2024-03-28 07:32:50  
2          0  2024-02-01 19:23:06  2024-01-28 14:47:11  
3          0  2024-02-13 13:50:51  2024-02-11 23:30:15  
4          0  2024-04-20 06:28:13  2024-04-10 07:08:52  
..       ...                  ...                  ...  
128        0  2023-10-15 19:52:06                 None  
129        0  2023-11-01 20:12:48                 None  
130        0  2023-10-20 17:53:20                 None  
131        0  2023-09-14 21:25:19                 None  
132        0  2023-09-28 10:40:52                 None  

[133 rows x 11 columns]

In [49]:
opinions.rating = opinions.rating.apply(lambda rate: rate.split("/")[0].replace(",",".")).astype(float)

## Podstawowe statystki

In [50]:
opinions_count = opinions.shape[0]
pros_count = opinions.pros.astype(bool).sum()
cons_count = opinions.cons.astype(bool).sum()
average_rating = opinions.rating.mean()

In [51]:
average_rating

4.845864661654136

In [ ]:
#opinions.rating
#opinions["rating"]